### PR TITLE
Update tasks to accept only JSON serializable data (bug 1174219)

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: lib/video/tasks.py [hunk 1]
```diff
@@ -18,8 +20,9 @@
 # Video decoding can take a while, so let's increase these limits.
 @task(time_limit=time_limits['hard'], soft_time_limit=time_limits['soft'])
 @set_modified_on
-def resize_video(src, instance, user=None, **kw):
+def resize_video(src, pk, user=None, **kw):
```
Comment (diox) [inline lib/video/tasks.py]: that `user` parameter is probably a not going to work either with JSON serialization

File: mkt/site/tasks.py [hunk 2]
```diff
@@ -40,15 +42,16 @@ def send_email(recipient, subject, message, real_email, from_email=None,
 
 
 @task
-def set_modified_on_object(obj, **kw):
+def set_modified_on_object(app_label, model_name, pk, **kw):
```
Comment (diox) [inline mkt/site/tasks.py]: pretty sure this doesn't actually need to be a task. Maybe that'd make things simpler if it wasn't?

File: lib/video/tasks.py (hunk 1 above)
Comment (robhudson) [inline lib/video/tasks.py]: Ah, good call out. I'll fix that.

File: mkt/site/tasks.py (hunk 2 above)
Comment (robhudson) [inline mkt/site/tasks.py]: I don't think this is bad as it is right now, do you?

Comment (diox) [inline mkt/site/tasks.py]: No, it's fine. We should have a bug about it, it's worth fixing (would make things simpler and reduce the amount of tasks in the queue)

**Comment by diox** [comment]
r+

**Pull request merged**